cloud-sql/mysql/client-side-encryption/src/main/java/com/example/cloudsql/EncryptAndInsertData.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.cloudsql;
+
+// [START cloud_sql_mysql_encrypt_insert]
+import com.google.crypto.tink.aead.AeadConfig;
+import com.google.crypto.tink.Aead;
+import com.google.crypto.tink.KeysetHandle;
+import com.google.crypto.tink.KmsClients;
+import com.google.crypto.tink.aead.AeadKeyTemplates;
+import com.google.crypto.tink.integration.gcpkms.GcpKmsClient;
+import com.google.crypto.tink.proto.KeyTemplate;
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import java.security.GeneralSecurityException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.util.Date;
+import javax.sql.DataSource;
+
+class EncryptAndInsertData {
+
+  public static void main(String[] args) throws GeneralSecurityException, SQLException {
+    // TODO(developer): Replace these variables before running the sample.
+    String dbUser = "your-username"; // e.g. "root", "mysql"
+    String dbPass = "your-password";
```

Comment:
Not sure it's a good idea to store the password in source code. Maybe replace this with a command-line argument?

Reply:
I think you're looking at an outdated version. The latest version reads this information in from environment variables.
https://github.com/GoogleCloudPlatform/java-docs-samples/pull/4821/files#diff-aab83b5ceecd4d6e83b84401f33a99c50b07b092e62493a6e31534a1019778adR35